packages/loot-core/src/server/accounts/rules.js

```diff
@@ -416,9 +444,22 @@ export class Action {
     this.options = options;
   }
 
-  exec(object) {
+  exec(object, transaction) {
     switch (this.op) {
       case 'set':
+        console.log("is set")
+        
+        if (typeof this.value == "string") {
+          let components = this.value.split("[|]")
+          if (components.length === 4 && components[0] === "regexp") {
+            let regex = new RegExp(components[2], 'ui');
+            let targetField = transaction[components[1]];
+            if (regex.test(targetField)){
+              object[this.field] =  targetField.replace(regex, components[3]);
```

Comment:
I think that this should update both the update object and the transaction so that multiple actions in a row will see the updated transaction.
```js
let updatedField = targetField.replace(regex, components[3]);
object[this.field] = updatedField;
transaction[this.field] = updatedField;
```